File: BUFF/BuffServer/src/main/java/vo/StockPoolConditionVO.java
package vo;

import po.StockPoolConditionPO;
import stockenum.StockPool;

import java.util.Set;

/**
 * Created by slow_time on 2017/3/24.
 */
public class StockPoolConditionVO {

    /**
     * 股票池的类型
     */
    public StockPool stockPool;
    /**
     * 所属板块的集合
     */
    public Set<String> block;
    /**
     * 所属行业的集合
     */
    public Set<String> industry;
    /**
     * 是否排除ST股票
     */
    public boolean excludeST;


    //无参数初始化方法
    public StockPoolConditionVO() {
        //TODO delete
        System.out.println(stockPool==null);
    }

    public StockPoolConditionVO(StockPool stockPool, Set<String> block, Set<String> industry, boolean excludeST) {
        this.stockPool = stockPool;
        this.block = block;
        this.industry = industry;
        this.excludeST = excludeST;
    }

    public StockPoolConditionVO(StockPoolConditionPO stockPoolConditionPO) {

        this.stockPool = stockPoolConditionPO.getStockPool();
        this.block = stockPoolConditionPO.getBlock();
        this.industry = stockPoolConditionPO.getIndustry();
        this.excludeST = stockPoolConditionPO.isExcludeST();
    }
}

File: BuffTreasureWebApp/vo/StockPoolConditionVO.js
/**
 * Created by slow_time on 2017/5/9.
 */
exports.StockPoolConditionVO = class {
    /**
     * 构造函数
     * adjust by wsw  去掉全部  改成随机500  用于测试  或者提供使用
     * @param stockPool     String 股票池，暂定包括（"随机500"，"沪深A股"，"中小板"，"创业板"，"自选股票池"）
     * @param benches       Array  板块列表，列表里的值类型为String，只有当stockPool为"自选股票池"时，才会有值，否则为null
     * @param industries    Array  行业列表，列表里的值类型为String，只有当stockPool为"自选股票池"时，才会有值，否则为null
     * @param excludeST     Bool   是否排除ST股，排除为true，不排除为false
     */
    constructor (stockPool, benches, industries , excludeST ) {
        this.stockPool = stockPool;
        this.benches = benches;
        this.industries = industries;
        this.excludeST = excludeST;
    }
};